src/main/java/com/emc/object/s3/jersey/S3JerseyClient.java

```diff
@@ -200,14 +202,27 @@ public S3JerseyClient(S3Config config, ClientHandler clientHandler) {
             }
         }
 
+        ClientHandler handler = client.getHeadHandler();
```

Comment:
need a comment here explaining that we have to re-order the filters and why it is necessary

Reply:
Added comments.